File: bestia-zoneserver/src/main/java/net/bestia/messages/internal/FromClient.java
package net.bestia.messages.internal;

import net.bestia.messages.AccountMessage;

/**
 * Message coming from the client sometimes are redundant and thus need to me
 * marked to they can get routed the right way.
 */
public class FromClient extends AccountMessage {

	private static final long serialVersionUID = 1L;
	private final AccountMessage payload;

	public FromClient(AccountMessage payload) {
		super(payload.getAccountId());

		this.payload = payload;
	}

	public Object getPayload() {
		return payload;
	}

	@Override
	public AccountMessage createNewInstance(long accountId) {
		return new FromClient(payload.createNewInstance(accountId));
	}

}

File: bestia-zoneserver/src/main/kotlin/net/bestia/messages/FromClient.kt
package net.bestia.messages

/**
 * Message coming from the client sometimes are redundant and thus need to me
 * marked to they can get routed the right way.
 * @Deprecated Use Envelope system.
 */
@Deprecated("Use Envelope System now.")
class FromClient(private val payload: AccountMessage) : AccountMessage(payload.accountId) {

  fun getPayload(): Any {
    return payload
  }

  override fun createNewInstance(accountId: Long): AccountMessage {
    return FromClient(payload.createNewInstance(accountId))
  }

  companion object {
    private const val serialVersionUID = 1L
  }
}
